### PR TITLE
Refactor polling + Poll only if window Visible

### PR DESCRIPTION
**Pull request by anxolin**
# Summary

Iterates over PR https://github.com/cowprotocol/cowswap/pull/1913 to refactor a bunch of places where we use polling. 

This PR creates a new hook: `usePolling`

This hook will periodically invoke a polling function and ensure the polling is not done while the browser tab/window is not visible.

## Motivation
While addressing this comment from Leandro https://github.com/cowprotocol/cowswap/pull/1913#pullrequestreview-1267805562 I saw how many repetitions we have in the polling logic. So, I gave it a try to use refactor into a new hook. 

## Testing
Testing instructions are coming soon

### CODE REVIEW

File: src/custom/state/price/updater.ts [hunk 1]
```diff
@@ -236,43 +237,28 @@ export default function FeesUpdater(): null {
...
+      const thereIsPreviousPrice = !!quoteInfo?.price?.amount
+      const isPriceRefresh = quoteInfo
+        ? thereIsPreviousPrice && quoteUsingSameParameters(quoteParams, quoteInfo)
+        : false
+
+      setLastUnsupportedCheck(Date.now())
+
+      refetchQuote({
+        quoteParams,
+        fetchFee: true, // TODO: Review this, because probably now doesn't make any sense to not query the feee in some situations. Actually the endpoint will change to one that returns fee and quote together
```
Comment (alfetopito) [inline src/custom/state/price/updater.ts:257]: I guess this no longer applies.
But no need to change it in this PR as the change is unrelated

**Review by alfetopito — APPROVED** [review]
Code looks ok, just some minor comments

**Comment by elena-zh** [comment]
Hey  @anxolin , can we close this?

**Comment by anxolin** [comment]
yes @elena-zh I will close and will give it another try here https://github.com/cowprotocol/cowswap/pull/2956, still WIP since its a bit delicate logic
